modRlp/src/org/aion/rlp/CompactEncoder.java

```diff
@@ -113,37 +113,23 @@
         }
         int oddlen = (nibbles.length - terminator) % 2;
         int flag = 2 * terminator + oddlen;
+
+        ByteArrayOutputStream buffer = new ByteArrayOutputStream();
```

Comment:
I would suggest using `ByteBuffer` here instead, seems to perform much better than a full stream
http://java-performance.info/java-io-bytearrayoutputstream/